src/status_im/subs/wallet/wallet.cljs

```diff
@@ -173,19 +173,19 @@
  :<- [:wallet/current-viewing-account]
  :<- [:wallet/network-details]
  (fn [[account networks] [_ query]]
-   (let [tokens          (map (fn [token]
-                                (assoc token
-                                       :networks           (utils/network-list token networks)
-                                       :total-balance      (utils/total-token-units-in-all-chains token)
-                                       :total-balance-fiat (utils/calculate-balance-for-token token)))
-                              (:tokens account))
-         sorted-tokens   (sort-by :name compare tokens)
-         filtered-tokens (filter #(or (string/starts-with? (string/lower-case (:name %))
-                                                           (string/lower-case query))
-                                      (string/starts-with? (string/lower-case (:symbol %))
-                                                           (string/lower-case query)))
-                                 sorted-tokens)]
-     filtered-tokens)))
+   (let [tokens        (map (fn [token]
+                              (assoc token
+                                     :networks           (utils/network-list token networks)
+                                     :total-balance      (utils/total-token-units-in-all-chains token)
+                                     :total-balance-fiat (utils/calculate-balance-for-token token)))
+                            (:tokens account))
+         sorted-tokens (sort-by :name compare tokens)]
+     (if query
+       (let [query-string (string/lower-case query)]
+         (filter #(or (string/starts-with? (string/lower-case (:name %)) query-string)
+                      (string/starts-with? (string/lower-case (:symbol %)) query-string))
```

Comment:
Kinda worried about what we mentioned today in our chat, About clojure.string functions not playing nice with nils.

Reply:
This code is now making sure that `query-string` is not `nil` (that is the change I'm adding here: avod performing this fitler when there's no query string), so we shouldn't worry, but I can use `some->` to add an extra layer of security.